MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+        if (areaDrops) {
+            // ... go through the file, and construct areas/drops as appropriate
+            for (NSDictionary *areaDrop in areaDrops) {
+                //This is getting complicated, so I'm going to move this out to a separate method.
+                [self addAreaDropFromDictionary:areaDrop toRegion:region];
+            }
+        }
+    }
+    
+    // We should now try to save the manged object context
+    if (![self didSuccessfullySaveContext]) {
+        NSLog(@"WARNING: Failed to save the item context!");
+    }
+}
+
+- (void)addAreaDropFromDictionary:(NSDictionary *)areaDropDict toRegion:(Region *)region
+{
+    //First, let's check if the area that this areaDrop takes place in actually exists!
+    NSString *areaCombinedName = [NSString stringWithFormat:@"%@-%@-%@",
+                                  region.name,
+                                  areaDropDict[@"area"],
+                                  areaDropDict[@"rank"]];
+    Area *area = [self getUniqueAreaWithCombinedName:areaCombinedName];
+    // If the area doesn't already exist...
+    if (!area) {
+        // ... We need to create it, and add it to this region
+        NSEntityDescription *newAreaDescription = [NSEntityDescription entityForName:@"Area" inManagedObjectContext:self.managedObjectContext];
+        area = [[Area alloc] initWithEntity:newAreaDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        area.name = areaDropDict[@"area"];
+        area.combinedName = areaCombinedName;
+        area.region = region;
+        area.rank = areaDropDict[@"rank"];
+        [region.areaSet addObject:area];
+    }
+    // If it wasn't already, the area is now attached to the region
+    
+    // We should now consider adding this drop to the area.
+    // First, check if the area drop already exists, by way of checking the idDecimalString
+    NSString *idDecimalString = areaDropDict[@"_id"];
+    AreaDrop *drop = [self getUniqueAreaDropWithDecimalID:idDecimalString];
+    //If the drop already exists...
+    if (drop) {
+        // ... do nothing. (At least for now).
+        return;
+    } else {
+        // This else is not needed, but helps for clarity IMO.
+        // Hook up the drop.
+        NSEntityDescription *newDropDescription = [NSEntityDescription entityForName:@"AreaDrop" inManagedObjectContext:self.managedObjectContext];
+        drop = [[AreaDrop alloc] initWithEntity:newDropDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        drop.idDecimalString = idDecimalString;
+        drop.method = areaDropDict[@"method"];
+        NSString *percent = areaDropDict[@"chance"];
+        drop.percentValue = percent.integerValue;
+        drop.quantity = areaDropDict[@"number"];
+        drop.rank = areaDropDict[@"rank"];
+        
+        // Hook up the area relationship.  We already ensured the area exists.
+        drop.area = area;
+        [area.dropSet addObject:drop];
```

Comment:
You shouldn't need to set this in both directions—setting it in one direction should automatically set the other direction if you've got the inverse relationship set up.